metis-authentication/metis-authentication-service/src/test/java/eu/europeana/metis/authentication/dao/TestPsqlMetisUserDao.java

```diff
@@ -0,0 +1,351 @@
+package eu.europeana.metis.authentication.dao;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import eu.europeana.metis.authentication.user.MetisUser;
+import eu.europeana.metis.authentication.user.MetisUserAccessToken;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.hibernate.Criteria;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.TransactionException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InOrder;
+import org.mockito.Mockito;
+
+/**
+ * @author Simon Tzanakis (Simon.Tzanakis@europeana.eu)
+ * @since 2017-11-03
+ */
+public class TestPsqlMetisUserDao {
+
+  private static SessionFactory sessionFactory;
+  private static Session session;
+  private static Transaction transaction;
+  private static Query query;
+  private static PsqlMetisUserDao psqlMetisUserDao;
+
+  @BeforeClass
+  public static void prepare() throws IOException {
+    sessionFactory = Mockito.mock(SessionFactory.class);
+    psqlMetisUserDao = new PsqlMetisUserDao(sessionFactory);
+    psqlMetisUserDao.setAccessTokenExpireTimeInMins(10);
+
+    session = Mockito.mock(Session.class);
+    transaction = Mockito.mock(Transaction.class);
+    query = Mockito.mock(Query.class);
+  }
+
+  @After
+  public void cleanUp() {
+    Mockito.reset(session);
+    Mockito.reset(transaction);
+    Mockito.reset(query);
+  }
+
+  @Test
+  public void createMetisUser() {
+    when(sessionFactory.openSession()).thenReturn(session);
+    when(session.beginTransaction()).thenReturn(transaction);
```

Comment:
I would move those to a separate method or even to prepare because they are called in the beginning of every test

Reply:
when(session.beginTransaction()).thenReturn(transaction); is not in all methods.
I will move when(sessionFactory.openSession()).thenReturn(session); to an before method.